skbuild/setuptools_wrap.py

```diff
@@ -474,8 +474,33 @@ def setup(*args, **kw):  # noqa: C901
     # specified
     if sys.platform == 'darwin':
         if plat_name is None:
-            # The following code is duplicated in bdist_wheel.finalize_options()
-            plat_name = "macosx-10.6-x86_64"
+            # If cmake osx deployment args are set then set plat from them
+            user_set = False
+            if cmaker.has_cmake_cache_arg(
+                    cmake_args, 'CMAKE_OSX_DEPLOYMENT_TARGET'):
+                # The loop here allows for cli flags to overload
+                # those set in the setup file.
+                for s in cmake_args:
+                    if 'CMAKE_OSX_DEPLOYMENT_TARGET' in s:
+                        version = s.split('=')[1]
+                user_set = True
+            else:
+                version = '10.6'
```

Comment:
Now-a-days, I think `10.9` is more common, per

https://github.com/MacPython/wiki/wiki/Spinning-wheels